sources/pagerduty-source/src/pagerduty.ts

```
@@ -181,6 +186,21 @@ export class Pagerduty {
     });
   }
 
+  async *getTeams(): AsyncGenerator<Team> {
+    let res;
+    try {
+      res = await this.client.get(`/teams`);
+    } catch (err) {
+      res = err;
```

Should we at least log the error if we are not going to fail the source?

---

I'm not familiar with the API (nor JS really) - got this far by copying `getPrioritiesResource` below. Happy to log error but don't want to vary too far from what already exists here. If logging is required, can you point me to an example somewhere?

---

yeah, let it fail. Replace with 
```
async *getTeams(): AsyncGenerator<Team> {
   const res = await this.client.get(`/teams`);
   for (const item of res.resource) {
        yield item;
   }
}
```

---

i've updated per your comments above to add pagination.
should i go with those changes or should i leave it as
```
async *getTeams(): AsyncGenerator<Team> {
   const res = await this.client.get(`/teams`);
   for (const item of res.resource) {
        yield item;
   }
}
```

i'm not sure the tests really cover these changes. are you able to test this connector locally?